### PR TITLE
Document non-default property values

### LINKED ISSUE #808: Document non-default kafka property values
A community user complained that one Kafka property auto_create_topics_enabled is set to true in Kafka but is set to false in Redpanda to protect against errant applications (e.g. KSQL) from spamming the system with new topics. 

There are probably a few cases where opinionated configuration choices like this should be called out to users migrating from Kafka.

We should collect the set of properties where our defaults differ from Kafka and add that as a section to: https://docs.redpanda.com/docs/platform/development/kafka-clients/

### PR DESCRIPTION
**Pull request by shorgi**
Fixes #808 

- New [Non-default property values](https://deploy-preview-1605--redpanda-documentation.netlify.app/docs/develop/kafka-clients/#non-default-property-values) section

### CODE REVIEW

File: docs/develop/kafka-clients.mdx [hunk 1]
```diff
@@ -45,3 +45,13 @@ Redpanda does not currently support the following Apache Kafka features:
 * Record deletion.
 
 If you have any issues while working with a Kafka tool, you can [file an issue](https://github.com/redpanda-data/redpanda/issues/new).
+
+## Non-default property values
```
Comment (shorgi) [inline docs/develop/kafka-clients.mdx:49]: SMEs - please review for correctness and any properties missing from this table.

File: docs/develop/kafka-clients.mdx [hunk 2]
```diff
@@ -45,3 +45,13 @@ Redpanda does not currently support the following Apache Kafka features:
...
+
+Redpanda sets default values for the following properties that are different than Kafka's default values for corresponding properties:
```
Comment (Feediver1) [inline docs/develop/kafka-clients.mdx:51]:
```suggestion
Redpanda sets default values for the following properties, which are different from Kafka's default values for corresponding properties:
```

**Review by Feediver1 — APPROVED** [review]

File: docs/develop/kafka-clients.mdx [hunk 3]
```diff
@@ -45,3 +45,13 @@ Redpanda does not currently support the following Apache Kafka features:
...
+
+| Property | Redpanda default value | Kafka default value | Notes |
+| :-- | :-- | :-- | :-- |
+|[`auto_create_topics_enabled`](../../reference/cluster-properties#auto_create_topics_enabled)| `false` | `true` | Prevents errant applications from creating too many new topics. | 
+|[`log_segment_ms`](../../reference/cluster-properties#log_segment_ms)| 14 days | 7 days | Sets the range of time within a segment that Redpanda can handle out-of-order timestamps. |
```
Comment (jcsp) [inline docs/develop/kafka-clients.mdx:56]: The description for log_segment_ms doesn't match the property:
```
 , log_segment_ms(
      *this,
      "log_segment_ms",
      "Default log segment lifetime in ms for topics which do not set "
      "segment.ms",
      {.needs_restart = needs_restart::no,
       .example = "3600000",
       .visibility = visibility::user},
      std::chrono::weeks{2},
      {.min = 60s})

```

Comment (shorgi) [reply docs/develop/kafka-clients.mdx:56]: @jcsp @VladLazar - the description in the table for `log_segment_ms` is meant to convey the change in its default value (from 7 days to 14 days) made in https://github.com/redpanda-data/redpanda/pull/8509. How do you recommend we revise or remove this description from the table?

**Review by jcsp — APPROVED** [review]

**Pull request merged**